File: java/src/main/java/mx/iteso/adapter/GoogleAdapter.java
package mx.iteso.adapter;

import mx.iteso.GoogleDocument;
import mx.iteso.utils.Format;
import mx.iteso.utils.Image;
import mx.iteso.utils.MSLicense;

/** GoogleAdapter class. */
public class GoogleAdapter implements IWordDocument {

    /** google document attribute. */
    private IGoogleDoc googleDocument;

    /** Google adapter constructor.
     * @param newGoogleDocument google document.
    */
    public GoogleAdapter(final GoogleDocument newGoogleDocument) {
        this.googleDocument = newGoogleDocument;
    }

    /** Get format.
     * @return format.
     */
    public Format getFormat() {
        return this.googleDocument.getStyle();
    }

    /** Get background.
     * @return background.
     */
    public Image getBackground() {
        return this.googleDocument.getBackground().getImage();
    }

    /** Set office version.
     * @param msOfficeVersion office version.
    */
    public void setMSOfficeVersion(final float msOfficeVersion) {
        return;
    }

    /** Get license.
     * @return license.
     */
    public MSLicense getLicense() {
        return this.googleDocument.getSharingPermissions() == 1
        ? new MSLicense(1) : new MSLicense(-1);
    }

    /** Is invalid.
     * @return is valid.
     */
    public boolean restrictEditIfLicenseIsInvalid() {
        return !this.getLicense().isValid();
    }
}

File: csharp/Adapter/Clases/GoogleAdapter.cs
namespace Adapter.Clases
{
    public class GoogleAdapter : IWordDocument {

        private readonly GoogleDoc _doc;

        private readonly MsLicense _msLicense;

        private float _msVersion;

        
        public GoogleAdapter(GoogleDoc doc1) {
            _doc = doc1;
            _msLicense = new MsLicense("Google license");
        }

        public Format GetFormat() {
            return new Format(_doc.GetFont());
        }

        public Image GetBackground() {
            var backgroundImage = _doc.GetBackground();
            return new Image(backgroundImage.GetImage().GetUrl());
        }

        public void SetMsOfficeVersion(float msOfficeVersion) {
            _msVersion = msOfficeVersion;
        }

        public float GetMsOfficeVersion() {
            return _msVersion;
        }

        public MsLicense GetLicense() {
            return _msLicense;
        }

        public bool RestrictEditIfLicenseIsInvalid(MsLicense msLic) {
            return _msLicense.GetLicense() == msLic.GetLicense();
        }
    }
}